### DISCUSSION TITLE
v3.4.0

### DISCUSSION THREAD

**Discussion by gnikit** [krvajal/vscode-fortran-support#677] [Announcements]
# What's New (PRE RELEASE) 

## 🎉 Modern Fortran Release of v3.3 🎉! 

- [Run and Debug individual files](#run-and-debug-individual-files)
- [LFortran Linter support](#lfortran-linter-support)
- [Log Channel improvements](#log-channel-improvements)
  - [Colourized Logs](#colourized-logs)
  - [Setting verbosity level](#setting-verbosity-level)
- [Improved Installation of dependencies using VS Code Tasks](#improved-installation-of-dependencies-using-vs-code-tasks)
- [`fypp` linting support](#fypp-linting-support)
- [Improved linter diagnostics](#improved-linter-diagnostics)
- [Performance Improvements](#performance-improvements)
  - [Improved performance of the linter](#improved-performance-of-the-linter)
- [Update native symbol provider](#update-native-symbol-provider)
- [Added options](#added-options)
  - [Hide Release Notes](#hide-release-notes)
- [Added Fortran Logo icon](#added-fortran-logo-icon)

## Run and Debug individual files

You can now run and debug individual files. This is useful for debugging small snippets of code. To do this, right-click on the file and select `Run File` or `Debug File` from the context menu.

## LFortran Linter support

The LFortran linter is now available in the extension. It can be enabled by setting the `fortran.linter.compiler` setting to `lfortran`.

## Log Channel improvements

The `Modern Fortran` log channel has had a small revamp.

### Colourized Logs

Logs are now colorised to make reading them easier

![image](https://user-images.githubusercontent.com/16143716/190214202-0cce3ee4-80da-4f4a-88bb-c36abd09c8f6.png)

### Setting verbosity level

You can now choose the verbosity level of the extension by setting the following option in the settings

```json
{
  "fortran.logging.level": "Error"
}
```

## Improved Installation of dependencies using VS Code Tasks

The extension dependencies are now installed using Visual Studio Code's Tasks. That means that the commands are run from within the VS Code terminal, inheriting any environment variables already present. Particularly useful when using Python virtual environments.

## `fypp` linting support

Adds some initial support for `fypp` when using `gfortran`. More compilers will follow soon!

![image](https://user-images.githubusercontent.com/16143716/190215420-085043a5-8250-4777-a0f0-7a94ec740987.png)

## Improved linter diagnostics

Add support for parsing plain text diagnostics from `gfortran` v11+
thus allowing the display of multiline diagnostics

```fortran
module err_mod
    private
    implicit none   ! <- Error here previously not shown
    contains
        subroutine foo(arg1, arg2)
            integer, intent(in) :: arg1, arg2
            print*, 'arg1:', arg1, 'arg2:', arg2
        end subroutine foo
        subroutine proc_with_err()
            call foo()
        end subroutine proc_with_err
end module err_mod
```

![image](https://user-images.githubusercontent.com/16143716/190216547-324a70c3-df57-4c13-800d-efbd57b52562.png)

## Performance Improvements

Some additional performance improvements have been made to the extension.

### Improved performance of the linter

Converted the linter into using asynchronous processes, which should improve the overall performance and responsiveness of the diagnostics.

## Update native symbol provider

The native symbol provider (one used when `fortls`) is not present) has been updated
to use the new VS Code API.

### Hide Release Notes

Hide release notes when the extension in being installed or updated.

```json
"fortran.notifications.releaseNotes": true
```

## Added Fortran Logo icon

Added a new icon for the Fortran Language files

![image](https://user-images.githubusercontent.com/16143716/192509653-609b25c2-f1bb-4370-ba2d-2781d7505814.png)

**Full Changelog**: https://github.com/fortran-lang/vscode-fortran-support/compare/v3.2.0...v3.4.0

<hr /><em>This discussion was created from the release <a href='https://github.com/fortran-lang/vscode-fortran-support/releases/tag/v3.4.0'>v3.4.0</a>.</em>

**Comment by awvwgk** [comment]
Is it useful to have a discussion thread for the release here and on discourse?

**Reply by gnikit** [reply]
The audiences between the 2 are not necessarily the same, but in all honesty it was more muscle memory rather than a conscious decision to start a discussion. FYI one thing that GitHub Discussions have over the Discourse is integration with the rest of the GitHub repo, like creating issues from a discussion.

**Comment by vgovze** [comment]
I haven't seen anyone comment on discourse.

the pre-release version has problems finding the fortls binary.
The current release version (v3.2) doesn't have an issue here.
Has the configuration or anything else changed in this regard?
I have updated the fortls to 2.13.0, the problem persists.

![image](https://user-images.githubusercontent.com/32483007/193764992-82d38269-d60e-40cc-bc62-654e1c6a132b.png)

Kind regards

**Reply by gnikit** [reply]
Could you provide the following:

What OS are you using?
Is fortls in your PATH?
What does the Log channel show when you set it in Debug mode?

**Reply by gnikit** [reply]
@vgovze Issue fixed in https://github.com/fortran-lang/vscode-fortran-support/pull/694 will issue a pre-release tonight

**Reply by vgovze** [reply]
Thank you for the quick response. I'll let you know if everything is working after the update.

For other diagnostic info, if it is still going to be of use. I am running Windows 10 and
no, the fortls is not in my path. I have provided only the path to the binary in the VScode
settings, which seems to have worked so far now.

I hope this is what you require from the log (the output panel in VScode -> Modern Fortran).
![image](https://user-images.githubusercontent.com/32483007/194228005-63078437-3efe-47eb-90e5-ab29f7eba58f.png)

**Reply by gnikit** [reply]
Have a try now, there is a pre-release out now. For future ease of use I added a Nightly build for the extension that runs every day Monday-Friday at 00:00 that will publish a pre-release for users to download. I think the issue should now be fixed.

**Reply by vgovze** [reply]
I've tried and it works in Pre-release version v3.4.2022100613. This issue is most probably resolved, thank you.

But in version v3.4.2022100702 of Pre-release (current), the fortls doesn't even start. I think this is a new
issue.

This is the entire output I get from the output panel from the ModernFortran extension.
![image](https://user-images.githubusercontent.com/32483007/194530031-bb517022-2167-479b-867b-42537bc912a2.png)

After trying to restart the language server from the command palette (Fortran: Restart the language server)
this error message pops out.
![image](https://user-images.githubusercontent.com/32483007/194530135-6b9cb5a4-e7c4-4e35-85c1-8e3421f66a92.png)
A similar error message pops up when running (Fortran: What's new).

**Reply by gnikit** [reply]
You can install the previous pre-release version if your press the arrow next to the Uninstall button.
The issue that you are encountering is only on Windows on UNIX platforms I did not encounter it. It appears that the extension does not activate properly I will have a look.

**Reply by vgovze** [reply]
It is now working fine. Thank you.
